docs/bsp.md

```diff
@@ -902,14 +1072,63 @@ trait TestReport {
 }
 ```
 
-The `TestReport` notification will be sent as the test execution of build targets completes.
+This request may trigger a compilation or other tasks on the selected build targets.
 
-This request may trigger a compilation on the selected build targets. The server is free to send any
-number of `build/compileReport`, `build/publishDiagnostics` and `build/logMessage` notifications
-during compilation before completing the response.
+The server may send any number of `build/task*`, `build/publishDiagnostics` and `build/logMessage` notifications
+to communicate about tasks triggered by the request to the client.
+
+#### Test Unit Notifications
 
-The client will get a `originId` field in the `TestReport` or `TestResult` if the `originId` field
-in the `TestParams` is defined.
+The server may inform about individual test units in task notifications that reference the originating task in their `taskId`.
+For example, the server can send a `taskStart`/`taskFinish` for each test suite in a target, and likewise for each test in the suite. 
+Individual test start notifications should specify `test-started` in the `dataKind` field and include the `TestStart` object and
+test finish notifications should specify `test-finished` in the `dataKind` field and include the `TestFinish` object in the `data` field.
+
+```scala
+trait TestStart {
+  /** Name or description of the test. */
+  def description: String
+  
+  /** Source location of the test, as LSP location. */
+  def location: Option[Location]
+}
+
+trait TestFinish {
+  /** Name or description of the test. */
+  def description: String
```

Comment:
Two questions:
  
1. Should this be named `description`? I think we should find a better name.
2. Should this be typed as `String`? Why not use a URI and call it id?

Reply:
I was thinking this should be whatever the UI displays. Could be method name, fqn or some string describing a requirement, depending on how test framework organizes its tests. Ideas on the name?

Reply:
We use `displayName` in `buildTarget`, so I propose we're consistent with that and rename it accordingly.